### PR TITLE
fix(cli): `wing test` returns 0 exit code upon failure

### LINKED ISSUE #2916: Wing test returns 0 exit code
### Feature Spec

Wing test command should have appropriate exit codes based on the outcome of the command. Test failures have a non zero exit code, as well as a compilation failure. Right now everything exits with a 0 exit code.

### Use Cases

CI, developer experience

### Implementation Notes

_No response_

### Component

CLI

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by eladb**
Fixes #2916 by returning exit code 0 when tests fail.

## Misc

Tweaks to a bunch of tests.

## Checklist

- [x] Title matches [Winglang's style guide](https://docs.winglang.io/contributing/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [x] Docs updated (only required for features)
- [x] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Comment by ShaiBer** [comment]
Happy days :) and looks good!
Will leave it up to @staycoolcall911  to approve

File: tools/hangar/src/utils.ts [hunk 1]
```diff
@@ -25,18 +25,13 @@ export async function runWingCommand(options: RunWingCommandOptions) {
       env: options.env,
     }
   );
-  if (options.expectStdErr) {
+
+  if (options.expectFailure) {
     expect(out.exitCode).not.toBe(0);
-    expect(out.stderr).not.toBe("");
   } else {
-    if (
```
Comment (staycoolcall911) [inline tools/hangar/src/utils.ts:32]: This change looks correct, but IIRC this was added by @yoav-steinberg to test panic in a consistent way. Is that still working after you remove this `if` statement and reverse the `expectFailure` flags in `error.test.ts` and `invalid.test.ts`?

**Review by staycoolcall911 — APPROVED** [review]
Left a question about 😱, but other than that looks great!

File: tools/hangar/src/utils.ts (hunk 1 above)
Comment (eladb) [reply tools/hangar/src/utils.ts:32]: I believe it works.

**Pull request merged**